### PR TITLE
Add MIMA to build

### PR DESCRIPTION
**Pull request by wsargent**
Fixes https://github.com/lightbend/ssl-config/issues/91

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -4,7 +4,12 @@ addSbtPlugin("com.typesafe.sbt"      % "sbt-osgi"            % "0.9.2")
...
-addSbtPlugin("com.typesafe.tmp"      % "sbt-header"          % "1.5.0-JDK6-0.1")
 addSbtPlugin("com.jsuereth"          % "sbt-pgp"             % "1.1.0")
 addSbtPlugin("com.lightbend.paradox" % "sbt-paradox"         % "0.2.13")
 addSbtPlugin("com.eed3si9n"          % "sbt-doge"            % "0.1.5")
+addSbtPlugin("com.typesafe"          % "sbt-mima-plugin"     % "0.3.0")
+
+
+// XXX is this really needed since we don't use JDK 1.6 any more?
+// why not use https://github.com/sbt/sbt-header ?
+addSbtPlugin("com.typesafe.tmp"      % "sbt-header"          % "1.5.0-JDK6-0.1")
```
Comment (wsargent) [inline project/plugins.sbt]: I think this is a one off project to create headers for a 1.6 build

Comment (marcospereira) [reply project/plugins.sbt]: Hum, I think we can leave this outside of this PR scope and then have another one to replace this with `sbt-header` (including any change at the headers format).

WDYT?

File: build.sbt [hunk 2]
```diff
@@ -20,6 +20,7 @@ lazy val sslConfigCore = project.in(file("ssl-config-core"))
   .settings(osgiSettings: _*)
   .settings(
     name := "ssl-config-core",
+    mimaPreviousArtifacts := Set("com.typesafe" %% "ssl-config-core" % "0.2.3"), // "sbt mimaReportBinaryIssues"
```
Comment (marcospereira) [inline build.sbt]: Needs to be empty for Scala 2.13:

https://travis-ci.org/lightbend/ssl-config/jobs/417340678#L1126

**Comment by ktoso** [comment]
Let's try to assign milestones when closing tickets and PRs :) I fixed it here and for others

**Comment by marcospereira** [comment]
I've assigned 0.2.4 here (it was not assigned before).

Thanks for the reminder, @ktoso.

**Comment by wsargent** [comment]
Is semantic versioning?

**Pull request merged**